File: android/delivery-api/GetTaxonomyGroup.java
// DocSection: delivery_api_get_taxonomy_group
// Tip: Find more about JavaRx SDK at https://docs.kontent.ai/androidandroid

// Prepares the DeliveryService configuration object
DeliveryOptions options = DeliveryOptions.builder()
    .projectId("<YOUR_PROJECT_ID>")
    .build();

// Initializes a DeliveryClient for Java projects
DeliveryClient client = new DeliveryClient(options);

// Gets a taxonomy group using a simple request
CompletionStage<TaxonomyGroup> personas = client.getTaxonomyGroup("personas");

// Gets a taxonomy group using RxJava
Observable.fromCompletionStage(personas)
    .subscribe(new Observer<TaxonomyGroup>() {
        @Override
        public void onSubscribe(@NonNull Disposable d) {
        }

        @Override
        public void onNext(@NonNull TaxonomyGroup taxonomyGroup) {
            TaxonomyGroup personasTaxonomyGroup = taxonomyGroup;
        }

        @Override
        public void onError(@NonNull Throwable e) {
        }

        @Override
        public void onComplete() {
        }
    });
// EndDocSection

File: ios/delivery-api/GetTaxonomyGroup.swift
// DocSection: delivery_api_get_taxonomy_group
// Tip: Find more about Swift SDKs at https://kontent.ai/learn/ios
import KenticoKontentDelivery

let client = DeliveryClient.init(projectId: "<YOUR_PROJECT_ID>")

client.getTaxonomyGroup(taxonomyGroupName: "personas", completionHandler: { (isSuccess, deliveryItem, error) in
   if isSuccess {
       if let taxonomyGroup = deliveryItems.item {
        // use your taxonomy group here
       }
    } else {
         if let error = error {
             print(error)
         }
    }
})
// EndDocSection